lib/find.ts

```diff
@@ -11,28 +11,49 @@ interface FileHandler {
   (path: FilePath, stats: Stats): void;
 }
 
-export async function find(src: string): Promise<[FilePath[], FilePath[]]> {
+interface PathExcluder {
+  (path: FilePath): boolean;
+}
+
+export async function find(
+  src: string,
+  exclude: string[],
+): Promise<[FilePath[], FilePath[]]> {
   const fileResults: FilePath[] = [];
   const archiveResults: FilePath[] = [];
 
-  await traverse(src, async (path: FilePath, stats: Stats) => {
-    if (!isSupportedSize(stats.size)) {
-      return;
-    }
+  await traverse(
+    src,
+    async (path: FilePath, stats: Stats) => {
+      if (!isSupportedSize(stats.size)) {
+        return;
+      }
 
-    if (isArchive(path)) {
-      archiveResults.push(path);
-      return;
-    }
+      if (isArchive(path)) {
+        archiveResults.push(path);
+        return;
+      }
 
-    fileResults.push(path);
-  });
+      fileResults.push(path);
+    },
+    (path: FilePath) => {
+      return exclude.includes(relative(src, path));
```

Comment:
I'm a bit skeptic to this, since string operations might be expensive. But it's better than io-operations on the disk.